codeserver/ubi9-python-3.9/run-code-server.sh

```diff
@@ -16,6 +16,31 @@ fi
 # Initilize access logs for culling
 echo '[{"id":"code-server","name":"code-server","last_activity":"'$(date -Iseconds)'","execution_state":"running","connections":1}]' > /var/log/nginx/vscode.access.log
 
+# Check if code-server exists
+if [ ! -f "/opt/app-root/src/.local/share/code-server" ]; then
+
+    # Check internet connection
+    if curl -Is http://www.google.com | head -n 1 | grep -q "200 OK"; then
+        # Internet connection is available
+        echo "Internet connection available. Installing specific extensions."
+
+        # Install specific extensions
+        code-server --install-extension ${SCRIPT_DIR}/utils/ms-python.python-2023.14.0.vsix
+        code-server --install-extension ${SCRIPT_DIR}/utils/ms-toolsai.jupyter-2023.3.100.vsix
+    else
+        # No internet connection
+        echo "No internet connection. Installing all extensions."
+
+        # Install all extensions
+        code-server --install-extension ${SCRIPT_DIR}/utils/ms-python.python-2023.14.0.vsix
+        code-server --install-extension ${SCRIPT_DIR}/utils/ms-toolsai.jupyter-2023.3.100.vsix
```

Comment:
Hi: The documentation says: The Jupyter Extension will automatically install the following extensions by default to provide enhanced Jupyter notebook experiences in VS Code.
    [Jupyter Keymap](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter-keymap) - to provide Jupyter-consistent keymaps
    [Jupyter Notebook Renderers](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.jupyter-renderers) - to provide renderers for MIME types such as latex, plotly, vega, etc.
    [Jupyter Cell Tags](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.vscode-jupyter-cell-tags) and [Jupyter Slide Show](https://marketplace.visualstudio.com/items?itemName=ms-toolsai.vscode-jupyter-slideshow) - to provide the ability to tag cells in notebooks and support for presentations
    
I guess this means that in ms.toolsai.jupyter extension vsix offline install mode, that still needs explicit one by one install of those extra 4 extensions like here, correct?

Reply:
Exactlym that's why we also provide them for use in disconnected environments [here](https://github.com/opendatahub-io/notebooks/blob/main/codeserver/ubi9-python-3.9/run-code-server.sh#L30)

Reply:
yup, thank you for the clarification on what "automatically install" means in disconnected environments.